File: backend/app/schemas/session.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class OrganizationSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    slug: str
    role: str


class SessionResponse(BaseModel):
    user_id: UUID
    display_name: str
    email: str
    organizations: list[OrganizationSummary]


class ActiveOrganizationRequest(BaseModel):
    organization_id: UUID


class ActiveOrganizationResponse(BaseModel):
    organization: OrganizationSummary

File: frontend/src/api/session.ts
export type OrganizationSummary = {
  id: string;
  name: string;
  slug: string;
  role: string;
};

export type Session = {
  user_id: string;
  display_name: string;
  email: string;
  organizations: OrganizationSummary[];
};

const apiUrl = import.meta.env.VITE_API_URL ?? "http://localhost:8000";

export async function fetchSession(accessToken: string): Promise<Session> {
  const response = await fetch(`${apiUrl}/api/v1/me`, {
    headers: { Authorization: `Bearer ${accessToken}` },
  });
  if (!response.ok) {
    throw new Error(response.status === 401 ? "Session expirée" : "API NeighborLink indisponible");
  }
  return response.json() as Promise<Session>;
}
